### PR TITLE
improve github PR template

### PR DESCRIPTION
**Pull request by dragonly**
### What problem does this PR solve? 
Improve the github PR template

### What is changed and how does it work?
Changed the github PR template, which encourages committer to write more understandable PR description.

Hopefully this will make people in the community feels easier to engage in the development process :wink:

### Check List 

Tests 

- No code

Code changes

None

Side effects

None

Related changes

- Need to cherry-pick to the release branch

### Does this PR introduce a user-facing change?:

```release-note
NONE
```

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,43 +1,59 @@
...
-- Breaking backward compatibility
+- [ ] Breaking backward compatibility <!-- If this PR breaks things deployed with previous TiDB Operator versions, check this box -->
+- [ ] Other side effects: <!-- Any other side effects, such as requiring additional storage / consumes substantial memory / potential reconcilation latency -->
 
-Related changes
+### Related changes
 
-- Need to cherry-pick to the release branch
-- Need to update the documentation
+- [ ] Need to cherry-pick to the release branch <!-- If this PR should also appear in the current release branch, check this box -->
```
Comment (csuzhangxc) [inline .github/pull_request_template.md:47]: we have two release branches, how about given two checkboxes for both versions?

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,43 +1,59 @@
 <!--
-Thank you for contributing to TiDB Operator! Please read TiDB Operator's [CONTRIBUTING](https://github.com/pingcap/tidb-operator/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
```
Comment (csuzhangxc) [inline .github/pull_request_template.md:2]: How about still keep the link to `CONTRIBUTING.md`?

File: .github/pull_request_template.md (hunk 1 above)
Comment (dragonly) [reply .github/pull_request_template.md:47]: Then this template must be changed every time we add a new release branch
Is there a better way of doing this? 🤔

File: .github/pull_request_template.md (hunk 2 above)
Comment (dragonly) [reply .github/pull_request_template.md:2]: kept

File: .github/pull_request_template.md (hunk 1 above)
Comment (csuzhangxc) [reply .github/pull_request_template.md:48]: I prefer to change this template for every release (we know the PR should be cherry-picked to which branch(es)). but still waiting for opinions from other reviewers.

Comment (dragonly) [reply .github/pull_request_template.md:48]: I don't think change template with code release is a good idea.
Maybe we could make the committer write the release number?
WDYT @lonng @DanielZhangQD

Comment (lonng) [reply .github/pull_request_template.md:48]: I think it's enough to let the author of PR marks whether this PR need to be cherry-picked, and the reviewer add some labels for dedicate release branches.

**Review by lonng — APPROVED** [review]
LGTM

**Review by csuzhangxc — APPROVED** [review]
LGTM

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,43 +1,60 @@
...
+Thank you for contributing to TiDB Operator!
+Please complete the following template before creating a PR :D
```
Comment (dragonly) [inline .github/pull_request_template.md:3]:
```suggestion
Please complete the following template before creating a PR
```

Comment (DanielZhangQD) [inline .github/pull_request_template.md]:
```suggestion
Please complete the following template before creating a PR.
```

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,43 +1,60 @@
...
+Ref: TiDB Operator [CONTRIBUTING](https://github.com/pingcap/tidb-operator/blob/master/CONTRIBUTING.md) document
 -->
 
-### What problem does this PR solve? <!--add and issue link with summary if exists-->
+### What problem does this PR solve?
+<!--
+Please describe the problem AS DETAILED AS POSSIBLE.
+Add and ISSUE LINK WITH SUMMARY if exists.
```
Comment (DanielZhangQD) [inline .github/pull_request_template.md:10]:
```suggestion
Add an ISSUE LINK WITH SUMMARY if exists.
```

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,43 +1,60 @@
...
+- [ ] Breaking backward compatibility <!-- If this PR breaks things deployed with previous TiDB Operator versions, check this box -->
+- [ ] Other side effects: <!-- Any other side effects, such as requiring additional storage / consumes substantial memory / potential reconciliation latency -->
 
-Related changes
+### Related changes
 
-- Need to cherry-pick to the release branch
-- Need to update the documentation
+- [ ] Need to cherry-pick to the release branch <!-- If this PR should also appear in the current release branch, check this box -->
+- [ ] Need to update the documentation <!-- If this PR introduces new features / change previous usages, check this box -->
```
Comment (DanielZhangQD) [inline .github/pull_request_template.md:49]:
```suggestion
- [ ] Need to update the documentation <!-- If this PR introduces new features or changes previous usages, check this box -->
```

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -1,43 +1,60 @@
 <!--
-Thank you for contributing to TiDB Operator! Please read TiDB Operator's [CONTRIBUTING](https://github.com/pingcap/tidb-operator/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
+Thank you for contributing to TiDB Operator!
+Please complete the following template before creating a PR
```
Comment (dragonly) [inline .github/pull_request_template.md:3]:
```suggestion
Please complete the following template before creating a PR.
```

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Pull request merged**